packages/datetime/src/dateInput.tsx

```diff
@@ -287,6 +291,23 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         Utils.safeInvoke(this.props.onChange, date === null ? null : fromMomentToDate(momentDate));
     }
 
+    private shouldCheckForDateChanges(prevMomentDate: moment.Moment, nextMomentDate: moment.Moment) {
+        return nextMomentDate !== null && !isMomentNull(prevMomentDate) && prevMomentDate.isValid();
```

Comment:
Copied this from the code above, but definitely seems worth changing. Will do. (https://github.com/palantir/blueprint/pull/1092/files#diff-2899fd9fc14d599c6e4062f791d99120L279)